subsys/bluetooth/mesh/cfg.c

```diff
@@ -87,6 +99,16 @@ int bt_mesh_gatt_proxy_set(enum bt_mesh_feat_state gatt_proxy)
 		return -ENOTSUP;
 	}
 
+	if (gatt_proxy == BT_MESH_FEATURE_DISABLED &&
+		feature_get(BT_MESH_GATT_PROXY) == BT_MESH_FEATURE_ENABLED) {
+
+		struct bt_mesh_subnet *sub = bt_mesh_subnet_find(check_proxy_adv, NULL);
+
+		if (!sub) {
+			bt_mesh_adv_update();
+		}
+	}
```

Comment:
If you move this `if` below line 115, you can get away with just 
```c
if (gatt_proxy == BT_MESH_FEATURE_DISABLED && !bt_mesh_subnet_find(node_id_is_running, NULL)) {
    bt_mesh_adv_update();
}
```

This would also guarantee that we get no race conditions between the adv-thread and this one, where the adv thread does `adv_update()` before we've updated `bt_mesh.flags` to the new state

Reply:
I agree with race conditions  (I doubted about them) but this simple condition will not work since multiple successful disabling is allowed according to implementation.

Reply:
Wouldn't that cause the `feature_set` function to return `-EALREADY`?

Reply:
you are right, my mistake